File: rust/src/models/quote_event_data.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{customer_id::CustomerId, quote_id::QuoteId, subscription_id::SubscriptionId};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct QuoteEventData {
    pub customer_id: CustomerId,

    pub quote_id: QuoteId,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub subscription_id: Option<SubscriptionId>,
}

impl QuoteEventData {
    pub fn new(customer_id: CustomerId, quote_id: QuoteId) -> Self {
        Self {
            customer_id,
            quote_id,
            subscription_id: None,
        }
    }
}

File: typescript/src/models/quoteEventData.ts
// this file is @generated
import { type CustomerId, CustomerIdSerializer } from "./customerId";
import { type QuoteId, QuoteIdSerializer } from "./quoteId";
import { type SubscriptionId, SubscriptionIdSerializer } from "./subscriptionId";

export interface QuoteEventData {
  customerId: CustomerId;

  quoteId: QuoteId;

  subscriptionId?: SubscriptionId | null;
}

export const QuoteEventDataSerializer = {
  _fromJsonObject(object: any): QuoteEventData {
    return {
      customerId: CustomerIdSerializer._fromJsonObject(object["customer_id"]),
      quoteId: QuoteIdSerializer._fromJsonObject(object["quote_id"]),
      subscriptionId:
        object["subscription_id"] != null
          ? SubscriptionIdSerializer._fromJsonObject(object["subscription_id"])
          : undefined,
    };
  },

  _toJsonObject(self: QuoteEventData): any {
    return {
      customer_id: CustomerIdSerializer._toJsonObject(self.customerId),
      quote_id: QuoteIdSerializer._toJsonObject(self.quoteId),
      subscription_id:
        self.subscriptionId != null
          ? SubscriptionIdSerializer._toJsonObject(self.subscriptionId)
          : undefined,
    };
  },
};
